api/app/src/command/medical/document/document-query.ts

```diff
@@ -138,6 +155,38 @@ export const updateConversionProgress = async ({
     };
     await PatientModel.update(updatedPatient, { where: patientFilter, transaction });
 
+    // START TODO 785 remove this once we're confident with the flow
+    const maxUpdateRetry = 3;
+    let curAttempt = 1;
+    while (curAttempt++ < maxUpdateRetry) {
+      const patientPost = await getPatientOrFail({
+        id: patient.id,
+        cxId: patient.cxId,
+        transaction,
+      });
+      log(
+        `[txn attempt ${curAttempt}] Status post-update: ${JSON.stringify(
+          patientPost.data.documentQueryProgress
+        )}`
+      );
+      if (
+        !isDocumentQueryProgressEqual(documentQueryProgress, patientPost.data.documentQueryProgress)
+      ) {
+        log(`[txn attempt ${curAttempt}] Status post-update not expected... trying again`);
```

Comment:
nit: could add a variable to indicate whether it will run again or not, so we could conditionally print `trying again` - helps with debugging. Could even use that variable on the while to have the logic in one place.

Reply:
To clarify, this message just denotes that we will try to update the doc query again, which always happens on the next line - not that we'll be looping again.

Updated message accordingly.